protocols/imap/src/test/java/org/apache/james/imap/decode/parser/SetAnnotationCommandParserTest.java

```diff
@@ -0,0 +1,88 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.imap.decode.parser;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import org.apache.james.imap.api.ImapCommand;
+import org.apache.james.imap.decode.ImapRequestStreamLineReader;
+import org.apache.james.imap.message.request.SetAnnotationRequest;
+import org.apache.james.mailbox.model.MailboxAnnotation;
+import org.apache.james.protocols.imap.DecodingException;
+import org.junit.Test;
+
+public class SetAnnotationCommandParserTest {
+
+    private static final String INBOX = "INBOX";
+    private static final MailboxAnnotation PRIVATE_ANNOTATION = MailboxAnnotation.newInstance("/private/comment", "This is my comment");
+    private static final MailboxAnnotation SHARED_ANNOTATION = MailboxAnnotation.newInstance("/shared/comment", "This one is for you!");
+    private static final MailboxAnnotation NIL_ANNOTATION = MailboxAnnotation.nil("/private/comment");
+    private SetAnnotationCommandParser parser = new SetAnnotationCommandParser();
+    private ImapCommand command = ImapCommand.anyStateCommand("Command");
+
+    @Test
+    public void decodeMessageShouldReturnRequestContainsOneAnnotation() throws DecodingException {
+        SetAnnotationRequest request = createRequestFromStringCommand("INBOX (/private/comment \"This is my comment\") \n", null);
```

Comment:
It's not ok to do that.
First, we don't even see what is the call that is under test.
Second, if using the API hurts, then you have to fix the API, not write methods that hide complexity in your test